spacepy/toolbox/__init__.py

```diff
@@ -190,9 +195,11 @@ def hypot(*args):
     .. image:: ../../source/images/hypot_no_extension_speeds_3cases.png
     """
     if lib.have_libspacepy:
-        if len(args) == 1 and hasattr(args, 'ndim'): # it is an array
-                ans = lib.hypot_tb(args[0], np.product(args[0].shape))
-                return ans
+        if len(args) == 1 and isinstance(args[0], np.ndarray):  # it is an array
```

Comment:
Agreed, so you are in favor of this change? Or have a different line in mind? With the `np.require` just below this the input could be a list...

Reply:
Yes, I think the explicit type check is probably necessary, as much as I like to avoid it.

It does seem a bit weird that we're testing if it's an array, and then casting to an array if it isn't, but then not using the fast array code. Of course if we're looking at patching up this implementation we also need to evaluate in the light of `numpy.linalg.norm`...I suggest punting hypot enhancements (beyond this fix and the further tests for it) to another PR/issue.

Reply:
I think the decision on this was it's fine and punt any further hypot optimization to later.